flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/util/TestSignalHandler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Signal;
+
+/**
+ * This signal handler / signal logger is based on Apache Hadoop's org.apache.hadoop.util.SignalLogger.
+ *
+ * <p>This is a reduced version of {@link org.apache.flink.runtime.util.SignalHandler}.
```

Comment:
I'm wondering whether this reduction makes sense. Couldn't we just move the existing SignalHandler to flink-runtime to flinkcore, and register it in the test logger with....well no we can't because flink-test-utils-junit doesn't depend on flink-core, and everything else uses flink-test-utils-junit as a test dependency. So there is quite literally no place to put it.
Well that's annoying 💢

Reply:
Yes, I was annoyed by this as well!